### PR TITLE
[AJ-1310] Split up ImportData module

### PR DESCRIPTION
**Pull request by nawatts**
https://broadworkbench.atlassian.net/browse/AJ-1310

Taking inspiration from @jladieu's "prefactoring" PRs... before adding more code to ImportData, this moves it to a top level `import-data` folder and splits it up into more focused modules.

This does not make any functional changes.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -2,6 +2,7 @@
 # See https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#codeowners-syntax
 /src/analysis @DataBiosphere/broad-interactive-analysis
 /src/constants/datasets.js @DataBiosphere/data-explorer-eng
+/src/import-data/ @DataBiosphere/data-explorer-eng @DataBiosphere/analysisjourneys
```
Comment (nawatts) [inline .github/CODEOWNERS:5]: AJ owns some import data journeys.

File: src/import-data/protected-data-utils.ts
Comment (nawatts) [inline src/import-data/protected-data-utils.ts:1]: This was easy enough to convert to TS on the way.

**Review by calypsomatic — APPROVED** [review]

**Review by pshapiro4broad — APPROVED** [review]
This looks fine as far as I can tell. It looks like everything from the original files ended up somewhere in the new code. It'd be nice if sonar could confirm that test coverage is unchanged by this PR but I don't see a direct way to check that.

**Comment by nawatts** [comment]
I didn't delete any tests. And for what it's worth, test coverage will like improve as I continue to work on this for AJ-1310.

**Pull request merged**